reddit/stream.go

```diff
@@ -90,11 +90,93 @@ func (s *StreamService) Posts(subreddit string, opts ...StreamOpt) (<-chan *Post
 	return postsCh, errsCh, stop
 }
 
+// Comments streams comments from the specified subreddit.
+// It returns 2 channels and a function:
+//   - a channel into which new comments will be sent
+//   - a channel into which any errors will be sent
+//   - a function that the client can call once to stop the streaming and close the channels
+// Because of the 100 result limit imposed by Reddit when fetching posts, some high-traffic
+// streams might drop submissions between API requests, such as when streaming r/all.
+func (s *StreamService) Comments(subreddit string, opts ...StreamOpt) (<-chan *Comment, <-chan error, func()) {
+	streamConfig := &streamConfig{
+		Interval:       defaultStreamInterval,
+		DiscardInitial: false,
+		MaxRequests:    0,
+	}
+	for _, opt := range opts {
+		opt(streamConfig)
+	}
+
+	ticker := time.NewTicker(streamConfig.Interval)
+	commentsCh := make(chan *Comment)
+	errsCh := make(chan error)
+
+	var once sync.Once
+	stop := func() {
+		once.Do(func() {
+			ticker.Stop()
+			close(commentsCh)
+			close(errsCh)
+		})
+	}
+
+	ids := set{}
+
+	go func() {
+		defer stop()
+
+		var n int
+		infinite := streamConfig.MaxRequests == 0
+
+		for ; ; <-ticker.C {
+			n++
+
+			comments, err := s.getComments(subreddit)
+			if err != nil {
+				errsCh <- err
+				if !infinite && n >= streamConfig.MaxRequests {
+					break
+				}
+				continue
+			}
+
+			for _, comment := range comments {
+				id := comment.FullID
+
+				// certain comment streams are inconsistent about the completeness of returned comments
+				// it's not enough to check if we've seen older comments, but we must check for every comment individually
+				if !ids.Exists(id) {
```

Comment:
This `ids` set should probably get cleaned up at some point for continuously-running streams, or else it could cause memory problems in long-running apps